### PR TITLE
Update styles in learn area left sidebar

### LINKED ISSUE #1678: Update Learn Area Left Sidebar Styles
This is an Outreachy Medium difficulty issue. You must have completed a simple issue (marked with label "outreachy") first, to be eligible for being assigned to this one.

Please only request to be assigned this issue if you're applying to the Dark Mode Outreachy project.

# Task

The Navigation in the left sidebar of the Learn area at https://ocaml.org/docs/data-types needs to be improved:

|Current Page|Planned Design|
|-|-|
|Screenshot 2023-10-18 at 17-37-10 Basic Datatypes and Pattern Matching · OCaml Tutorials|Screenshot 2023-10-18 at 17-37-10 Basic Datatypes and Pattern Matching · OCaml Tutorials|

Here is [the Figma document for reference regarding the styles](https://www.figma.com/file/J0w4b9exaHhVpvabKoGqTn/Design-Github-Issues?type=design&node-id=300%3A2&mode=dev&t=SVZcP2fuAQa8onKt-1).

How to solve this issue:

1. Assign Tailwind CSS styles in `learn_sidebar.eml` to match the Figma design. Try to match the design as closely as you can with default Tailwind spacings.

### PR DESCRIPTION
**Pull request by FatumaA**
This PR closes #1678.
It refactors the learn area as per [this design](https://www.figma.com/file/J0w4b9exaHhVpvabKoGqTn/Design-Github-Issues?type=design&node-id=306-213&mode=design&t=YHnMNNphjxcms7Du-0) by:

- Adding a left border and adjusts radius on current item in left side bar list.
- Adjusting visual hierarchy of the left side bar by indenting the sub items and removing bold classes from them. 
- Adding more space around the left side bar.

Changes are shown in the screenshot below: 

Screenshot 2023-10-19 at 00 54 35

### CODE REVIEW

**Comment by FatumaA** [comment]
This is ready for review @sabine

**Comment by FatumaA** [comment]
I noticed 3 issues while working on this, I'm not sure if they have previously been noted. 

1. The exercises side bar nav is looking like the rest except it does not seem to use the shared code and does not have the active class as a result. I think this is related to this open [PR](https://github.com/ocaml/ocaml.org/pull/1631) though.
2. In learn - docs/platform - the landing page content does not related to the side bar content untill you click on the side bar link. Illustrated below:

Screenshot 2023-10-19 at 15 29 40

Screenshot 2023-10-19 at 15 29 53

3. And the mobile right side nav content is not visible at the footer if the content is long. As shown in the screenshot below:

Screenshot 2023-10-19 at 15 32 04

**Comment by sabine** [comment]
@FatumaA you are correct, that other PR is making changes to the exercises sidebar.

It's a good observation that we don't have a sidebar entry on the platform page. :thinking: You could add a link "Overview" to the sidebar on the platform page that is active when we are on /docs/platform.

**Comment by FatumaA** [comment]
Ok I will do that. 

As for the footer blocking the sidebar on smaller sceeens.. should I attempt a fix as part of this PR or should it be considered a separate issue?

**Comment by sabine** [comment]
re 3. the sidebar should be scrollable with enough padding at the bottom.

The footer is already being addressed in https://github.com/ocaml/ocaml.org/issues/1591

File: src/ocamlorg_web/lib/handler.ml [hunk 1]
```diff
@@ -42,12 +42,11 @@ let learn_guides req =
   Dream.redirect req (Url.tutorial (List.hd tutorials).slug)
 
 let platform _req =
-  let tools = Data.Tool.all in
   let tutorials =
     Data.Tutorial.all
     |> List.filter (fun (t : Data.Tutorial.t) -> t.section = Platform)
   in
-  Dream.html (Ocamlorg_frontend.platform ~tutorials tools)
+  Dream.redirect _req (Url.tutorial (List.hd tutorials).slug)
```
Comment (sabine) [inline src/ocamlorg_web/lib/handler.ml]: We cannot redirect to the platform tutorials here. The platform tools page is a proper page that has its own merit and is in the correct location.

What you can do is add an entry to the sidebar that says "Overview" and links to the platform page.

Comment (FatumaA) [reply src/ocamlorg_web/lib/handler.ml]: Ah ok,
I will revert changes.

What I had done so far was to add a markdown entry for the content on the platform page. 

And was currently trying to see how I can access/translate the tools.yml file to json or a simple array or something along those lines to be able to display the parts of the original platform page that was manipulating that data

Comment (sabine) [reply src/ocamlorg_web/lib/handler.ml]: Indeed, converting the platform tools page to markdown would go too far at this point. It's likely the design on that page will be extended beyond what markdown can do.

Comment (FatumaA) [reply src/ocamlorg_web/lib/handler.ml]: good point. 
Will change tact then to what you've suggested.
Thank you.

**Comment by FatumaA** [comment]
@sabine I think its ok now

**Review by christinerose — APPROVED** [review]

File: src/ocamlorg_frontend/layouts/learn_layout.eml [hunk 2]
```diff
@@ -81,11 +81,11 @@ let sidebar_link
 ~current =
   match current with
   | true ->
-    <a class="flex text-primary-600 bg-primary-100 py-2 px-3 rounded-md text-sm font-semibold" href="<%s href %>">
+    <a class="flex text-primary-600 bg-primary-100 border-l-4 border-primary-600 py-4 px-3 rounded-sm font-bold leading-10" href="<%s href %>">
```
Comment (sabine) [inline src/ocamlorg_frontend/layouts/learn_layout.eml]:
```suggestion
    <a class="flex text-primary-600 bg-primary-100 border-l-4 border-primary-600 py-2 px-3 rounded-sm font-bold leading-6" href="<%s href %>">
```

File: src/ocamlorg_frontend/layouts/learn_layout.eml [hunk 3]
```diff
@@ -94,8 +94,8 @@ let sidebar_link_group
 title
 link_html
 =
-  <div class="space-y-2 flex flex-col">
-    <div class="text-sm font-semibold flex px-3 py-2 uppercase"><%s title %></div>
+  <div class="space-y-2 pl-2 pr-8 flex flex-col">
```
Comment (sabine) [inline src/ocamlorg_frontend/layouts/learn_layout.eml]:
```suggestion
  <div class="pl-2 pr-8 flex flex-col">
```

**Review by sabine — COMMENTED** [review]
Looking really good. The design is unfortunately not very precise with the paddings in this case, so I looked into this and found a few adjustments that will make it look overall more consistent wrt spacings/paddings.

In addition to the comments, I made, there's `platform.eml` line 10, as well as `tutorial.eml` line 16 where the gap should be reduced to `gap-5` instead of `gap-10`

**Comment by FatumaA** [comment]
> Looking really good. The design is unfortunately not very precise with the paddings in this case, so I looked into this and found a few adjustments that will make it look overall more consistent wrt spacings/paddings.
> 
> In addition to the comments, I made, there's `platform.eml` line 10, as well as `tutorial.eml` line 16 where the gap should be reduced to `gap-5` instead of `gap-10`

Somehow I missed this notification 🤦🏾‍♀️ Just made the requested changes. 

Also, would you be open to reviewing my final application?

**Comment by FatumaA** [comment]
This is ready @sabine

**Comment by sabine** [comment]
Great, thank you @FatumaA! I'm merging this in a moment.

I'm not an outreachy mentor, so I am not reviewing the outreachy applications. @gpetiot or @SaySayo may be able to help.

**Comment by FatumaA** [comment]
Thank you. 

It's ok, Sayo already did the review for me 🙂

**Pull request merged**